File: App/app/src/main/java/com/example/econo_misons/database/models/Budget.java
package com.example.econo_misons.database.models;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

//Model of the Budget Table
@Entity(tableName = "Budget")
public class Budget {
    @PrimaryKey(autoGenerate = true)
    @NonNull
    @ColumnInfo(name = "ID")
    public int id;

    @ColumnInfo(name = "NAME_BUD")
    public String budgetName;

    public Budget(String budgetName){ this.budgetName = budgetName;}

    @Override
    public String toString() {
        return "Budget{" +
                "id=" + id +
                ", budgetName='" + budgetName + '\'' +
                '}';
    }
}

File: src/src/main/java/com/example/testbare/database/entities/Budget.kt
package com.example.testbare.database.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey


@Entity(
    tableName = "Budget",
    foreignKeys = [
        ForeignKey(
            entity = Mois::class,
            parentColumns = arrayOf("mois_id"),
            childColumns = arrayOf("bud_mois_id"),
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Categorie::class,
            parentColumns = arrayOf("cat_categorie"),
            childColumns = arrayOf("bud_categorie"),
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Budget (
    @PrimaryKey(autoGenerate = true)
    val bud_id: Int,
    @ColumnInfo(name = "bud_montant")
    var bud_montant: Float,
    @ColumnInfo(name = "bud_mois_id", index = true)
    var bud_mois_id: Int,
    @ColumnInfo(name = "bud_categorie", index = true)
    var bud_categorie: String
)
